recipes/sdbus-cpp/all/conanfile.py

```diff
@@ -0,0 +1,108 @@
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class SdbusCppConan(ConanFile):
+    name = "sdbus-cpp"
+    license = "LGPL-2.1-or-later"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/Kistler-Group/sdbus-cpp"
+    description = "High-level C++ D-Bus library for Linux designed" \
+                  " to provide easy-to-use yet powerful API in modern C++"
+    topics = ("conan", "dbus", "sd-bus", "sdbus-c++", "sdbus-cpp")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_code_gen": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_code_gen": False,
+    }
+    generators = "pkg_config"
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("Only Linux supported")
+        if self.options.shared:
+            del self.options.fPIC
+        tools.check_min_cppstd(self, "17")
```

Comment:
This needs an if check and some boilerplate 

You can copy this https://github.com/conan-io/conan/issues/8002#issue-736257740